### PR TITLE
Docs: add new site markdown files

### PR DESCRIPTION
**Pull request by jackye1995**
@samredai @rdblue 

Second step for new documentation site migration, add contents:
- from `iceberg-docs/docs/content/docs` to `iceberg/documentation/versioned`
- from `iceberg-docs/landing-page/content/common` to `iceberg/documentation/common`

Also update labeler for documentation to point to `documentation/**/*`

### CODE REVIEW

**Comment by jackye1995** [comment]
Did a direct copy using:

```
cp -r ../iceberg-docs/docs/content/docs documentation/versioned
cp -r ../iceberg-docs/landing-page/content/common documentation/common
```

License missing for all redirect files:

```
 !????? /home/runner/work/iceberg/iceberg/documentation/common/project/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/api/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/api/javadoc/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/asf/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/community/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/community/blogs.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/community/join.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/community/talks.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/flink/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/format/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/format/spec.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/format/terms.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/integrations/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/project/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/project/benchmarks.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/project/how-to-release.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/project/roadmap.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/project/security.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/project/trademarks.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/releases/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/releases/release-notes.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/releases/0.12.1/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/releases/0.13.0/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/releases/latest/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/spark/_index.md
 !????? /home/runner/work/iceberg/iceberg/documentation/versioned/tables/_index.md
```

My view is that we can remove these files in the Iceberg repo, when we sync to `iceberg-docs` repo we do a `rsync` instead of `cp -r` to ensure existing files without change are not removed in `iceberg-docs`. Any thoughts?

@samredai

**Comment by samredai** [comment]
All of the markdown files that contain content can just have the license headers added right below the hugo front matter and it will be detected by the license check. That's also true for the `_index.md` files but hugo will then consider that a file with content and there are other implications like the section titles in the left nav will be hyperlinked but show an empty page.

We should be able to solve this by adding `*_index.md` to the [.rat-excludes](https://github.com/apache/iceberg/blob/master/dev/.rat-excludes) file. I think this is ok since those don't have any content in them anyway and just hold the section header names or an external link. Then adding the license header below the `---` front matter will allow those to pass the check.

**Comment by jackye1995** [comment]
Thanks for the suggestion! Yes I noticed the UI changes when trying to add license to those files, that's why I was suggesting to not include those files in the main repo.

But I did not know we can exclude files from license checks, that sounds like a much better solution, I will do that instead, thanks for the suggestion!

File: documentation/common/community/blogs.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+---
+url: blogs
+weight: 200
+---
```
Comment (kbendick) [inline documentation/common/community/blogs.md:4]: Does this file need to be in the rat excludes too because the license isn't at the start?

**Comment by jackye1995** [comment]
@samredai @rdblue @kbendick updated to use `docs` instead of `documentation` as folder. Let me know if there is anything else.

**Comment by samredai** [comment]
lgtm, thanks @jackye1995!

**Review by kbendick — APPROVED** [review]
+1. Thanks @jackye1995!

I haven't opened this locally, but at least the change from `documentation` -> `docs` I can say I looked through pretty closely.

Can run locally if needed.

**Comment by jackye1995** [comment]
@samredai @kbendick thanks for the review, I will merge it for now since it's blocking all the doc related updates. We can update anything missing in subsequent PRs if necessary.

**Pull request merged**